qcfractal/tests/test_compute.py

```diff
@@ -112,6 +112,94 @@ def test_task_client_restart(fractal_compute_server):
+    geometric_options = {
+        "keywords": None,
+        "qc_spec": {"driver": "gradient", "method": "cookiemonster", "basis": "", "keywords": None, "program": "rdkit"},
+    }
+
+    ret1 = client.add_compute("rdkit", "cookiemonster", "", "energy", None, [mol])
+    ret2 = client.add_procedure("optimization", "geometric", geometric_options, [mol])
+    fractal_compute_server.await_results()
+
+    base_ids = [ret1.submitted[0], ret2.submitted[0]]
+    old_tasks = client.query_tasks(base_result=base_ids)
+
+    # Regenerate, but old one exists. Should be a no-op
+    upd = client.modify_tasks("regenerate", base_result=base_ids)
+    assert upd.n_updated == 0
+    new_tasks = client.query_tasks(base_result=base_ids)
+
+    for old_task, new_task in zip(old_tasks, new_tasks):
+        assert old_task.status == "ERROR"
+        assert old_task.id == new_task.id
+        assert old_task.base_result == new_task.base_result
+        assert old_task.modified_on == new_task.modified_on
+        assert old_task.created_on == new_task.created_on
+
+    # Manually delete the old task
+    db = fractal_compute_server.objects["storage_socket"]
+    db.del_tasks([x.id for x in old_tasks])
+
+    # Actually deleted?
+    del_task = client.query_tasks(base_result=base_ids)
+    assert len(del_task) == 0
+
+    # Now regenerate
+    upd = client.modify_tasks("regenerate", base_result=base_ids)
+    new_tasks = client.query_tasks(base_result=base_ids)
+    assert upd.n_updated == 2
+    for old_task, new_task in zip(old_tasks, new_tasks):
+        assert new_task.status == "WAITING"
+        assert old_task.id != new_task.id  # Task ids must now be different
+        assert old_task.base_result == new_task.base_result
+        assert old_task.modified_on < new_task.modified_on  # New task must be newer
+        assert old_task.created_on < new_task.created_on  # New task must be newer
+
+    assert old_tasks[0].spec.args[0]["molecule"]["id"] == new_tasks[0].spec.args[0]["molecule"]["id"]
+    assert (
+        old_tasks[0].spec.args[0]["molecule"]["identifiers"]["molecule_hash"]
+        == new_tasks[0].spec.args[0]["molecule"]["identifiers"]["molecule_hash"]
+    )
+    assert old_tasks[1].spec.args[0]["initial_molecule"]["id"] == new_tasks[1].spec.args[0]["initial_molecule"]["id"]
+    assert (
+        old_tasks[1].spec.args[0]["initial_molecule"]["identifiers"]["molecule_hash"]
+        == new_tasks[1].spec.args[0]["initial_molecule"]["identifiers"]["molecule_hash"]
+    )
+
+    # The status of the result should be reset to incomplete
+    res = client.query_procedures(base_ids)
+    assert all(x.status == "INCOMPLETE" for x in res)
+
+
+def test_task_modify(test_server):
+    client = ptl.FractalClient(test_server)
+
+    # Add a single computation
+    mol = ptl.models.Molecule(**{"geometry": [0, 0, 1], "symbols": ["He"]})
+    ret1 = client.add_compute("rdkit", "cookiemonster", "", "energy", None, [mol], tag="test_tag_1", priority=1)
```

Comment:
I thought priorities could be one of `{"HIGH", "MEDIUM", "LOW"}`. Has this changed?

Reply:
https://github.com/MolSSI/QCFractal/blob/e6c108f17beb3c248d6f8302dd9a826f528d708f/qcfractal/interface/models/task_models.py#L40

It's an integer, but there is some conversions happening. Thanks for pointing this out, though, I have been making an effort to consistently use the enums themselves.